src/connector/codec/src/decoder/avro/mod.rs

```diff
@@ -33,28 +33,46 @@ use super::{bail_uncategorized, uncategorized, Access, AccessError, AccessResult
 use crate::decoder::avro::schema::avro_schema_to_struct_field_name;
 
 #[derive(Clone)]
-/// Options for parsing an `AvroValue` into Datum, with an optional avro schema.
+/// Options for parsing an `AvroValue` into Datum, with a root avro schema.
 pub struct AvroParseOptions<'a> {
-    /// Currently, this schema is only used for decimal.
-    ///
-    /// FIXME: In theory we should use resolved schema.
-    /// e.g., it's possible that a field is a reference to a decimal or a record containing a decimal field.
```

Comment:
Now we can support _reference to a decimal or a record containing a decimal field_.